utility/src/main/java/org/oppia/util/parser/ListTagHandler.kt

```diff
@@ -0,0 +1,173 @@
+package org.oppia.util.parser
+
+import android.content.Context
+import android.text.Editable
+import android.text.Html
+import android.text.Spannable
+import android.text.Spanned
+import android.text.Spanned.SPAN_EXCLUSIVE_EXCLUSIVE
+import android.text.Spanned.SPAN_MARK_MARK
+import android.text.style.LeadingMarginSpan
+import org.oppia.util.parser.ListTagHandler.ListTag
+import org.oppia.util.parser.StringUtils.LI_TAG
+import org.oppia.util.parser.StringUtils.OL_TAG
+import org.oppia.util.parser.StringUtils.UL_TAG
+import org.xml.sax.XMLReader
+import java.util.*
+
+/**
+ * Called when the HTML parser reaches an opening or closing tag.
+ * We only handle list tags and ignore other tags.
+ *
+ * <ul> and <ol> tags are pushed to the [lists] stack and popped when the closing tag is reached.
+ *
+ * <li> tags are handled by the [ListTag] instance corresponding to the parent tag.
+ *
+ * Reference: https://github.com/daphliu/android-spannable-list-sample/tree/master/app/src/main/java/com/daphneliu/sample/listspansample
```

Comment:
Sorry, I missed this before. We shouldn't be copying this class. Using it as a reference is fine, but that involves understanding the purpose of the class and then rewriting it from scratch without looking at the code of the original anymore. This class seems like a copy--please rewrite it. I suggest deleting this class, then writing it from scratch based on the contract (e.g. how this class is needed by other neighboring classes).

Ditto for any other classes added in this PR that are copies of classes from other repositories.

Reply:
Done.

Reply:
It seems the class was renamed, but it is otherwise the same. Did you rewrite it? I wouldn't expect the code to look the same. If you're rewriting it based on the needs of the class & not using the original code as a reference, I'd expect it to change.

Reply:
Instead of rewriting it, I changed implementation to simple class that checks the opening and closing tag and adds bullets and numbered list.